apps/arweave/src/ar_node_worker.erl

```diff
@@ -1926,3 +1710,317 @@ dump_mempool(TXs, MempoolSize) ->
+				usd_to_ar_rate = Rate,
+				scheduled_usd_to_ar_rate = ScheduledRate,
+				packing_2_5_threshold = 0,
+				strict_data_split_threshold = PrevB#block.strict_data_split_threshold,
+				hash_preimage = SolutionPreimage,
+				recall_byte = RecallByte1,
+				previous_solution_hash = PrevB#block.hash,
+				partition_number = PartitionNumber,
+				nonce_limiter_info = NonceLimiterInfo2,
+				poa2 = case PoA2 of not_set -> #poa{}; _ -> PoA2 end,
+				poa2_cache = PoA2Cache,
+				recall_byte2 = RecallByte2,
+				reward_key = element(2, RewardKey),
+				price_per_gib_minute = PricePerGiBMinute2,
+				scheduled_price_per_gib_minute = ScheduledPricePerGiBMinute2,
+				denomination = Denomination2,
+				redenomination_height = RedenominationHeight2,
+				double_signing_proof = may_be_get_double_signing_proof(PrevB, State),
+				merkle_rebase_support_threshold = MerkleRebaseThreshold,
+				chunk_hash = get_chunk_hash(PoA1, Height),
+				chunk2_hash = get_chunk_hash(PoA2, Height)
+			}, PrevB),
+			
+			BlockTimeHistory2 = lists:sublist(
+				ar_block:update_block_time_history(UnsignedB, PrevB),
+				?BLOCK_TIME_HISTORY_BLOCKS + ?STORE_BLOCKS_BEHIND_CURRENT),
+			BlockTimeHistory3 = lists:sublist(BlockTimeHistory2, ?BLOCK_TIME_HISTORY_BLOCKS),
+			UnsignedB2 = UnsignedB#block{
+				block_time_history = BlockTimeHistory2,
+				block_time_history_hash = ar_block:block_time_history_hash(
+						BlockTimeHistory3)
+			},
+			SignedH = ar_block:generate_signed_hash(UnsignedB2),
+			PrevCDiff = PrevB#block.cumulative_diff,
+			SignaturePreimage = << (ar_serialize:encode_int(CDiff, 16))/binary,
+					(ar_serialize:encode_int(PrevCDiff, 16))/binary, (PrevB#block.hash)/binary,
+					SignedH/binary >>,
+			Signature = ar_wallet:sign(element(1, RewardKey), SignaturePreimage),
+			H = ar_block:indep_hash2(SignedH, Signature),
+			B = UnsignedB2#block{ indep_hash = H, signature = Signature },
+			ar_watchdog:mined_block(H, Height, PrevH),
+			?LOG_INFO([{event, mined_block}, {indep_hash, ar_util:encode(H)},
+					{solution, ar_util:encode(SolutionH)}, {height, Height},
+					{step_number, StepNumber}, {steps, length(Steps)},
+					{txs, length(B#block.txs)},
+					{chunks,
+						case B#block.recall_byte2 of
+							undefined -> 1;
+							_ -> 2
+						end}]),
+			ar_block_cache:add(block_cache, B),
+			ar_events:send(solution, {accepted, #{ indep_hash => H, source => Source }}),
+			apply_block(update_solution_cache(H, Args, State));
+		_Steps ->
+			ar_events:send(solution,
+					{rejected, #{ reason => bad_vdf, source => Source }}),
+			?LOG_ERROR([{event, bad_steps},
+					{prev_block, ar_util:encode(PrevH)},
+					{step_number, StepNumber},
+					{prev_step_number, PrevStepNumber},
+					{prev_next_seed, ar_util:encode(PrevNextSeed)},
+					{output, ar_util:encode(NonceLimiterOutput)}]),
+			{noreply, State}
+	end.
+
+update_solution_cache(H, Args, State) ->
+	#{ solution_cache := Map, solution_cache_records := Q } = State,
+	Q2 = queue:in(H, Q),
```

Comment:
I think this line can move into the `false` branch below since it's not used in the `true` branch